### ISSUE TITLE
practice mode doesn't exit after team kill

### ISSUE THREAD

**Issue by def-** [ddnet/ddnet#4754]
cheeser0613 reports:

> old ddnet practice mode can be exit after team kill, but it wont in the latest server (idk is it a bug or intended, if its not a bug maybe add a new command to exit practice before start because the only way to exit now is to change team)

**Comment by def-** [comment]
> around 1-2months ago, i like to run some faily map with my friend
when we fail at a part, we will use practice mode to try on how to pass the map first before k and rerun
after make sure we know how to pass it then we will just kill (press K) then just run again, no extra move to exit the practice mode

> but in current server
if i do samething in the server (practice mode then kill and rerun) and some how reach the end
ill realise that i doesnt count as pass because in current server, the practice mode wont gone after you kill and not every player realise this changes
the current way to exit practice mode is to change team, not like before we can just kill and everything good to go

**Comment by Chairn** [comment]
I think it doesn't reset when one of the team mate hasn't crossed start line.

**Comment by C0D3D3V** [comment]
Since I'm about to create the new HUD (https://github.com/ddnet/ddnet/pull/3401) anyway. Could we also show in the HUD if the team is in practice mode. I could transfer the information to CClientData. I personally find it relatively intuitive how to exit practice mode. But a visualization would not hurt I think.

**Comment by C0D3D3V** [comment]
Currently, you can simply teleport directly to the destination without moving. I find this quite practical because you can explicitly practice special parts without having to do the noobfilter all the time.

**Comment by def-** [comment]
> Yes you can skip noob filters, but you can not finish after skipping, because you are in practice.

But with your `/exitpractice` they could finish.

**Comment by C0D3D3V** [comment]
No, because the whole team would be killed. And at the spwanpoint spawn again. Sorry maybe I have expressed myself badly. They would not stay in place, but it would be like if the team had just gone to the spwan in the team.

**Comment by C0D3D3V** [comment]
Currently you can't get out of practice mode if you haven't started the race, which you can't start again because you are in practice mode. With the `/exitpractice` command this behavior should be overridden and the whole team should be killed and spawn neutral in the spwanpoint. So that they can start a new race without practice mode.

**Comment by def-** [comment]
Ok then. Any other opinions?

**Comment by Chairn** [comment]
I fear this might bring more bugs and exploit than solve problems. Switching to a new team is currently the only solution.

**Comment by C0D3D3V** [comment]
I just noticed that the behavior is probably map parameter dependent. For example, on the novice map Just2Easy, the whole team is killed and thrown out of practice mode even if you were in practice before the start. On the Brutal Map Gridwyn, however, the behavior is as described above. I'll see if I can find out what the parameter is. The question is what behavior do we want now or should that continue to depend on a parameter?

**Comment by def-** [comment]
Huh? It should really be the same behavior, I'm not aware that there is a difference.

**Comment by C0D3D3V** [comment]
Ah I have made messes when testing xD sorry!
It is actually the same behavior. 
The difference is only if you made /tp before the start. Because if you do /tp before the start then you can no longer start the race and thus the whole team will stay in practice forever.

**Comment by KebsCS** [comment]
Might be resolved with the `/unpractice` command now, it's the intended way to disable practice
